pkg/cosmos/client_test.go

```diff
@@ -0,0 +1,85 @@
+package cosmos
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+)
+
+func TestPackageDescribe(t *testing.T) {
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("/package/describe", func(w http.ResponseWriter, req *http.Request) {
+		assert.Equal(t, "POST", req.Method)
+		assert.Equal(
+			t,
+			"application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v3",
+			req.Header.Get("Accept"),
+		)
+		assert.Equal(
+			t,
+			"application/vnd.dcos.package.describe-request+json;charset=utf-8;version=v1",
+			req.Header.Get("Content-Type"),
+		)
+		payload := map[string]string{}
+		err := json.NewDecoder(req.Body).Decode(&payload)
+		assert.NoError(t, err)
+		assert.Equal(t, "dcos-test-cli", payload["packageName"])
+
+		var pkgInfo PackageInfo
+
+		pkgInfo.Package.Resource.CLI.Plugins = map[string]map[string]*Plugin{
+			"linux": map[string]*Plugin{
+				"x86-64": &Plugin{
+					Kind: "zip",
+					URL:  "https://linux.example.com/dcos-test-cli.zip",
+				},
+			},
+			"darwin": map[string]*Plugin{
+				"x86-64": &Plugin{
+					Kind: "zip",
+					URL:  "http://darwin.example.com/dcos-test-cli.zip",
+				},
+			},
+			// Test case for https://jira.mesosphere.com/browse/COPS-3052.
```

Comment:
The COPS ticket is closed and cannot be opened by devs outside of Mesosphere, a description of the problem handling this test case check would be more useful.

Reply:
I removed this link and just put a simple comment as there is already a long description in the associated code.